library/kotlin/src/io/envoyproxy/envoymobile/stats/Counter.kt

```diff
@@ -8,4 +8,6 @@ interface Counter {
    * Increment the counter by the given count.
    */
   fun increment(count: Int = 1)
+
+  fun attach(tags: List<Tag>): Counter
```

Comment:
This method allows dynamically adding tags during runtime for clients.

Reply:
I would suggest having tags be an optional parameter to `increment` (and other "record" stats methods).

Reply:
I was discussing this with @jingwei99 during our 1:1. What I like about the current "attach" API is that the caller can hold the returned Counter and use it for subsequent increments rather than always having to pass down the tag list.

I can also see how it might be easier on the caller to call increment with tag lists and let the stats client deal with incrementing the correct counter rather than manually maintaining counter objects to select based on runtime tags. 

I could see how one might use either based on the situation.

Reply:
had a discussion over slack, will go with the suggestions and keep both apis, ty both :)